lib/ClangImporter/ImportType.cpp

```diff
@@ -472,7 +486,29 @@ namespace {
     }
 
     ImportResult VisitReferenceType(const clang::ReferenceType *type) {
-      return Type();
+      auto pointeeQualType = type->getPointeeType();
+      auto quals = pointeeQualType.getQualifiers();
+      Type pointeeType =
+          Impl.importTypeIgnoreIUO(pointeeQualType, ImportTypeKind::Value,
+                                   AllowNSUIntegerAsInt, Bridgeability::None);
+
+      if (pointeeQualType->isFunctionType()) {
+        return importFunctionPointerLikeType(type, pointeeType);
+      }
+
+      if (Impl.isOverAligned(pointeeQualType)) {
+        return importOverAlignedFunctionPointerLikeType(type, Impl);
+      }
+
+      PointerTypeKind pointerKind;
+      if (quals.hasConst()) {
+        pointerKind = PTK_UnsafePointer;
+      } else {
+        pointerKind = PTK_UnsafeMutablePointer;
+      }
+
+      return {pointeeType->wrapInPointer(pointerKind),
+              ImportHint::None};
```

Comment:
Why not `ImportHint::OtherPointer`?

Reply:
We return `ImportHint::None` in other branches too, plus returning `OtherPointer` here would result in wrapping the pointer in `Optional` (after converting the type we call `adjustTypeForConcreteImport` and from there `canImportAsOptional`).

Maybe the better than using `None` would be to create a separate `ImportHint` for C++ references and be explicit?

Reply:
Indeed, we don't want those side-effects of `OtherPointer`. I briefly looked at the kinds of special-casing that we do for pointer import hints, and none of them seem relevant to C++ references, so I agree, `ImportHint::None` is the correct choice.